### PR TITLE
packagers: fix compression settings

### PR DESCRIPTION
**Pull request by chouquette**
### Description

This fixes compression related properties, for which only the default value was ever used

### CODE REVIEW

File: lib/omnibus/packagers/deb.rb [hunk 1]
```diff
@@ -676,33 +676,42 @@ def remove(packages)
       shellout!("apt-get remove -y --force-yes #{packages}")
     end
 
-    def compression_level(val = nil)
-      unless val.nil?
+    def compression_level(val = NULL)
+      if null?(val)
+        @compression_level || 6
```
Comment (chouetz) [inline lib/omnibus/packagers/deb.rb:681]: Pardon my lack of ruby knowledge, not sure what the `||` does. Would `=` be more meaningful?
```suggestion
        @compression_level = 6
```

File: lib/omnibus/packagers/deb.rb [hunk 2]
```diff
@@ -676,33 +676,42 @@ def remove(packages)
       shellout!("apt-get remove -y --force-yes #{packages}")
     end
 
-    def compression_level(val = nil)
-      unless val.nil?
+    def compression_level(val = NULL)
+      if null?(val)
```
Comment (chouetz) [inline lib/omnibus/packagers/deb.rb]: Is there any specific reason you turned from `nil` to `NULL` (I think I understand the logic was reversed and it might be enough to change the behaviour, isn't it?)

File: lib/omnibus/packagers/deb.rb [hunk 3]
```diff
@@ -676,33 +676,42 @@ def remove(packages)
...
       end
-      @compression_level = val || 6
     end
     expose :compression_level
 
-    def compression_threads(val = nil)
-      unless val.nil?
+    def compression_threads(val = NULL)
+      if null?(val)
+        @compression_threads || 1
```
Comment (chouetz) [inline lib/omnibus/packagers/deb.rb:694]: If my above comment makes sense, otherwise you can dismiss them all
```suggestion
        @compression_threads = 1
```

File: lib/omnibus/packagers/deb.rb [hunk 4]
```diff
@@ -676,33 +676,42 @@ def remove(packages)
...
       end
-      @compression_threads = val || 1
     end
     expose :compression_threads
 
-    def compression_algo(val = nil)
-      unless val.nil?
+    def compression_algo(val = NULL)
+      if null?(val)
+        @compression_algo || "gzip"
```
Comment (chouetz) [inline lib/omnibus/packagers/deb.rb:709]:
```suggestion
        @compression_algo = "gzip"
```

File: lib/omnibus/packagers/rpm.rb [hunk 5]
```diff
@@ -834,13 +834,16 @@ def remove(packages)
       end
     end
 
-    def compression_level(val = nil)
-      unless val.nil?
+    def compression_level(val = NULL)
+      if null?(val)
+        @compression_level || 6
```
Comment (chouetz) [inline lib/omnibus/packagers/rpm.rb:839]:
```suggestion
        @compression_level = 6
```

File: lib/omnibus/packagers/rpm.rb [hunk 6]
```diff
@@ -849,20 +852,25 @@ def compression_level(val = nil)
     # This will be ignored when compression_algo is set to gzip as this isn't
     # supported by gzip
     #
-    def compression_threads(val = nil)
-      # Don't reject the value if compression_algo is != gzip in order to avoid
-      # forcing an ordering for the properties
-      unless val.nil?
+    def compression_threads(val = NULL)
+      if null?(val)
+        @compression_threads || 1
```
Comment (chouetz) [inline lib/omnibus/packagers/rpm.rb:857]:
```suggestion
        @compression_threads = 1
```

File: lib/omnibus/packagers/rpm.rb [hunk 7]
```diff
@@ -849,20 +852,25 @@ def compression_level(val = nil)
...
       end
-      @compression_threads = val || 1
     end
     expose :compression_threads
 
-    def compression_algo(val = nil)
-      unless val.nil?
+    def compression_algo(val = NULL)
+      if null?(val)
+        @compression_algo || "gzdio"
```
Comment (chouetz) [inline lib/omnibus/packagers/rpm.rb:872]:
```suggestion
        @compression_algo = "gzdio"
```

File: lib/omnibus/packagers/deb.rb (hunk 1 above)
Comment (chouquette) [reply lib/omnibus/packagers/deb.rb:681]: We're in the same boat, I don't understand ruby very much :sweat_smile: 

My understanding is that this function is a getter & setter. It will behave as a getter when the passed value is `NULL` (and/or `nil`? but other functions in the file seem to be using `NULL`)

So in this branch, we're returning a value, which is either `@compression_level` if it was set, or `6` as a default value.

The setter part is in the other branch, when `val` isn't `NULL`, in which case we assign it to `@compression_level`

File: lib/omnibus/packagers/deb.rb (hunk 2 above)
Comment (chouquette) [reply lib/omnibus/packagers/deb.rb]: Mostly consistency with the rest of the code, I honestly have no idea if it makes a difference

File: lib/omnibus/packagers/deb.rb (hunk 3 above)
Comment (chouquette) [reply lib/omnibus/packagers/deb.rb:694]: I *think* the current code is correct, but I so understand your question & doubt :grimacing:

File: lib/omnibus/packagers/deb.rb (hunk 1 above)
Comment (chouquette) [reply lib/omnibus/packagers/deb.rb:681]: After more testing and explanations from Alex, I switched back to nil, but have to stick to `@foo || default`, otherwise we assign the default value back everytime we fetch it, which defeats the purpose of having a setter

**Review by chouetz — APPROVED** [review]
